src/Cms/Cors.php

```diff
@@ -0,0 +1,236 @@
+<?php
+
+namespace Kirby\Cms;
+
+use Kirby\Exception\InvalidArgumentException;
+use Kirby\Toolkit\Str;
+
+/**
+ * Handles CORS (Cross-Origin Resource Sharing)
+ * headers for HTTP responses
+ *
+ * @package   Kirby Cms
+ * @author    Bastian Allgeier <bastian@getkirby.com>
+ * @link      https://getkirby.com
+ * @copyright Bastian Allgeier
+ * @license   https://getkirby.com/license
+ * @since     5.2.0
+ */
+class Cors
```

Comment:
`General docs comment` (placing this here to have a separate discussion thread):

Your documentation snippet is great and shows real-world config examples. What I'm missing is some warnings on the security impact:
- One general warning that enabling CORS can open the site to malicious requests from external sites (and should only be done if specifically needed and with a configuration that opens up as little as needed).
- A specific warning on the `allowCredentials` option (allows to execute actions with the permissions of the logged in user; option should only be enabled if the external origin is run under the dev's control and can fully be trusted).

Reply:
Thanks for the feedback. I've added a **Security Considerations** heading to track these concers – Do you think they're enough?

Reply:
Thanks, I think that's great. I've added two more sentences to make it even more clear for those without prior CORS experience.